pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/TextIndexHandler.java

```diff
@@ -135,17 +137,20 @@ public void updateIndices(SegmentDirectory.Writer segmentWriter, IndexCreatorPro
     }
     for (String column : columnsToAddIdx) {
       ColumnMetadata columnMetadata = _segmentDirectory.getSegmentMetadata().getColumnMetadataFor(column);
-      if (shouldCreateTextIndex(columnMetadata)) {
+      if (shouldCreateTextIndex(columnMetadata, _segmentDirectory.getSegmentMetadata())) {
         createTextIndexForColumn(segmentWriter, columnMetadata, indexCreatorProvider, indexCreatorProvider);
       }
     }
   }
 
-  private boolean shouldCreateTextIndex(ColumnMetadata columnMetadata) {
+  private boolean shouldCreateTextIndex(ColumnMetadata columnMetadata, SegmentMetadata segmentMetadata) {
     if (columnMetadata != null) {
       // Fail fast upon unsupported operations.
       checkUnsupportedOperationsForTextIndex(columnMetadata);
-      return true;
+
+      // Create text index only for those segments that are created within recent time-window
+      return segmentMetadata == null
+          || segmentMetadata.getIndexCreationTime() >= Instant.now().minusSeconds(3600).toEpochMilli();
```

Comment:
we don't want to touch older segments, not its indexes it can have any other indexes.
Should we use `getEndTime` of segment if we are going with time based index?

Reply:
Ideally, lets add config `buildIndexForOldSegments` as config. 
```
"fieldConfigList":[
  {
     "name":"text_col_1",
     "encodingType":"RAW",
     "indexTypes": ["TEXT"],
     "properties":{"fstType":"native"},
     "buildForOldSegements" : false
  }
  ``` 
  
We can modify the `IndexLoadingConfig.java`.
Here we have one method :
```
privatee boolean extractBuildForOldSegementsForTextIndexFromFieldConfig(TableConfig tableConfig) {

}
```
```
private boolean shouldCreateTextIndex () {
  checkUnsupportedOperationsForTextIndex(columnMetadata) && indexLoadingConfig.buildForOldSegements()
}
```